addons/l10n_account_withholding/models/account_payment.py

```diff
@@ -0,0 +1,64 @@
+from odoo import api, fields, models
+from odoo.exceptions import UserError
+
+
+class AccountPayment(models.Model):
+    _inherit = 'account.payment'
+
+    # ------------------
+    # Fields declaration
+    # ------------------
+
+    l10n_account_wth_line_ids = fields.One2many(
+        comodel_name='account.move.line',
+        string='Withholding Lines',
+        compute='_l10n_account_wth_compute_withholding_line_ids',
+    )
+
+    # --------------------------------
+    # Compute, inverse, search methods
+    # --------------------------------
+
+    @api.depends('move_id.line_ids')
+    def _l10n_account_wth_compute_withholding_line_ids(self):
+        for payment in self:
+            payment.l10n_account_wth_line_ids = payment.move_id.line_ids.filtered(lambda p: p.tax_line_id.l10n_account_wth_is_wth_tax)
+
+    @api.depends('payment_method_line_id')
+    def _compute_outstanding_account_id(self):
+        # EXTEND account to skip the computation when the context key is set.
+        keep_outstanding_accounts = self._context.get('keep_outstanding_account')
+        payment_to_skip = self.env['account.payment']
+        if keep_outstanding_accounts:
+            payment_to_skip = self.filtered('outstanding_account_id')
+        super(AccountPayment, self - payment_to_skip)._compute_outstanding_account_id()
+
+    # -----------------------
+    # CRUD, inherited methods
+    # -----------------------
+
+    @api.model_create_multi
+    def create(self, vals_list):
+        # EXTENDS account
+        # We need to provide an account at creation for the move to be correctly generated, but the computation would
+        # overwrite the account by the one on the payment method line. While this is a good thing if an account is set on them,
+        # it is an issue if it's empty.
+        return super(AccountPayment, self.with_context(keep_outstanding_account=True)).create(vals_list)
+
+    def write(self, vals):
+        # EXTENDS account
+        # These can be changed if needed without impacting the record.
+        # We cannot allow changes that would impact the lines directly, because there is no way to get the withholding.
+        # info back once the register payment wizard is closed.
+        allowed_fields = self._withholding_payment_fields_allowed_to_be_written()
+        significant_changes = any(field not in allowed_fields for field in vals.keys())
+        if significant_changes and self.l10n_account_wth_line_ids:
+            raise UserError(self.env._("You cannot modify payments that have withholding taxes on it.\n"
+                              "If it is required, please cancel the payment and register a new one."))
+        return super().write(vals)
+
+    @api.model
+    def _withholding_payment_fields_allowed_to_be_written(self):
+        """ Returns a set of fields which are allowed to be edited on an existing payment which includes withholding lines. """
+        return {'state', 'memo', 'partner_bank_id', 'is_sent'}
```

Comment:
So if you made a mistake, you need to delete the payment and start over?

Reply:
That is how it is done at the moment.